bento/src/main/java/com/yelp/android/bento/core/ViewPrecognitionCache.kt

```diff
@@ -0,0 +1,26 @@
+package com.yelp.android.bento.core
+
+// Cache of how many of each view holder type's were actually inflated.
+object ViewPrecognitionCache {
+    // This is a map of a name of a screen, to map of ViewHolder type's to how many view holder
+    // type's were requested.
+    @JvmStatic val requiredViewHolders = mutableMapOf<String, MutableMap<Any, Int?>?>()
+
+    // TODO make neater
+    fun incrementForViewHolder(name: String, viewHolderType: Any) {
```

Comment:
Instead of a name, would a hashCode of the recyclerview be sufficient instead? So it could be automatically applied to all? (Or, if no name, use a hashcode).

The idea being that what causes problem is often when we add and remove to many views to one single recyclerview.

It would not prevent memory bloating if we navigate between lots of recycler views though, but could be a good middle ground?

Reply:
We talked on Slack about why the hashCode won't work.